### PR TITLE
Enable testing arm64 Mac architecture in the CI.

### LINKED ISSUE #1054: [Feature]: Add arm64 Mac testing
### What would you like to see added to NeuroConv?

Just noticed that the testing matrix does not include `macos-latest` which uses arm64 (not intel) runners: https://github.com/catalystneuro/neuroconv/blob/main/.github/workflows/testing.yml#L23

### Is your feature request related to a problem?

_No response_

### Do you have any interest in helping implement the feature?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/catalystneuro/neuroconv/blob/main/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/catalystneuro/neuroconv/issues)?

### PR DESCRIPTION
**Pull request by h-mayorquin**
As in the title. Shoul fix #1054

Let's get the ball rolling on this.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
OK, this is working, we need somehow to skip the doc test for DeepLabCut with M1s and python=3.9. On the other hand, if we follow the numpy convention we should stop supporting python 3.9 now so maybe we don't need to do that.

File: src/neuroconv/datainterfaces/behavior/deeplabcut/_dlc_utils.py [hunk 1]
```diff
@@ -379,7 +379,9 @@ def add_subject_to_nwbfile(
     video_name, scorer = h5file.stem.split("DLC")
     scorer = "DLC" + scorer
 
-    df = _ensure_individuals_in_header(pd.read_hdf(h5file), individual_name)
+    data_frame_from_hdf5 = pd.read_hdf(h5file)
```
Comment (h-mayorquin) [inline src/neuroconv/datainterfaces/behavior/deeplabcut/_dlc_utils.py:382]:
```suggestion
```

**Comment by pauladkisson** [comment]
> But this also has the support for the intel mac which I think was only discontinued from pros two years ago or something? To be honest, I would prefer to test only on latest mac, the CI is already super bloated on my view but I think there might be a case here for the two.
> 
> How do you feel about it?

I am still using an intel mac, so I would like the tests to include it. Plus, I think they're still relatively common. I agree that the CI is a bit bloated and takes too long to pick up new jobs, but I don't think this is the place to make cuts.

**Comment by h-mayorquin** [comment]
>  but I don't think this is the place to make cuts.
This make sense. 

Let's brainstorm how to improve the CI at some point.

**Review by bendichter — APPROVED** [review]

**Pull request merged**